js/controller.js

```diff
@@ -567,6 +572,12 @@ module.exports = function(OO, _, $, W) {
       // add loadedmetadata event listener to main video element
       if (videoElement) {
         videoElement.addEventListener('loadedmetadata', this.metaDataLoaded.bind(this));
+
+        // add the AirPlay TargetAvailability event listener
+        if (window.WebKitPlaybackTargetAvailabilityEvent) {
+          videoElement.addEventListener('webkitplaybacktargetavailabilitychanged',
```

Comment:
Maybe you can also add another listener for when the media it's already on the appleTV or being casted.
I found this article related with that:
https://developer.apple.com/documentation/webkitjs/adding_an_airplay_button_to_your_safari_media_controls/#2940027
It's not mandatory...up to you to implement this.

Reply:
Great catch @marcosmx!